rockchip: add FriendlyARM NanoPC T4 support

msdos03 opened the pull request:
Hardware
--------
RockChip RK3399 ARM64 (6 cores)
4GB LPDDR3 RAM
1x 1000 Base-T
1 GPIO LED (status)
HDMI 2.0
3.5mm TRRS AV jack
Micro-SD slot
16GB eMMC
1x USB 3.0 Port
2x USB 2.0 Port
1x USB Type-C Port
1x M.2 PCI-E Port
AP6356S (BCM4356) SDIO WiFi & Bluetooth adapter
--------
Note: AP6356S is not supported yet due to the lack of firmware and NVRAM

Tested on NanoPC-T4 by me.
NanoPC-T4 only have a adc-button, kmod-gpio-button-hotplug will cause a kernel panic, so remove it.

1715173329 commented on target/linux/rockchip/image/armv8.mk:
```diff
@@ -12,15 +12,25 @@ define Device/firefly_roc-rk3328-cc
+  DEVICE_MODEL := NanoPC T4
+  SOC := rk3399
+  IMAGE/sysupgrade.img.gz := boot-common | boot-script | pine64-img | gzip | append-metadata
+  DEVICE_PACKAGES := kmod-phy-realtek
```
this is enable via target Kconfig directly

1715173329 commented on target/linux/rockchip/Makefile:
```diff
@@ -16,7 +16,7 @@ endef
 
 include $(INCLUDE_DIR)/target.mk
 
-DEFAULT_PACKAGES += uboot-envtools partx-utils e2fsprogs mkf2fs kmod-gpio-button-hotplug
```
It would be better to paste the panic log here and see if there's any fix could apply.

1715173329 commented on target/linux/rockchip/image/armv8.mk:
```diff
@@ -12,15 +12,25 @@ define Device/firefly_roc-rk3328-cc
   SOC := rk3328
   IMAGE/sysupgrade.img.gz := boot-common | boot-script | pine64-img | gzip | append-metadata
-  DEVICE_PACKAGES := kmod-usb-net-rtl8152
+  DEVICE_PACKAGES := kmod-usb-net-rtl8152  kmod-gpio-button-hotplug
```
```suggestion
  DEVICE_PACKAGES := kmod-usb-net-rtl8152 kmod-gpio-button-hotplug
```

And below

msdos03 commented:
@1715173329 
This is the first error message
```
[    4.514961] ------------[ cut here ]------------
[    4.515400] WARNING: CPU: 3 PID: 219 at include/linux/memcontrol.h:945 handle_mm_fault+0x218/0x22c
[    4.516211] Modules linked in: gpio_button_hotplug
[    4.516649] CPU: 3 PID: 219 Comm: rm Not tainted 5.15.118 #0
[    4.517157] Hardware name: FriendlyElec NanoPC-T4 (DT)
[    4.517616] pstate: 40400005 (nZcv daif +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
[    4.518238] pc : handle_mm_fault+0x218/0x22c
[    4.518627] lr : handle_mm_fault+0x11c/0x22c
[    4.519012] sp : ffff800009c839d0
[    4.519310] x29: ffff800009c839d0 x28: ffff00000282a4c0 x27: 0000000000000002
[    4.519951] x26: ffff000001b18ba8 x25: 0000000000000215 x24: ffff000001b18b40
[    4.520591] x23: 0000000096000045 x22: 0000000000000001 x21: 0000ffff93514401
[    4.521230] x20: 0000000000000200 x19: 0000000000000215 x18: 0000000000000000
[    4.521870] x17: 0000000000000000 x16: 0000000000000000 x15: 0000000000000000
[    4.522509] x14: 0000000000000000 x13: 0000000000000000 x12: 0000000000000000
[    4.523149] x11: 0000000000000000 x10: 0000000000000000 x9 : 0000000000000000
[    4.523788] x8 : 0000000000000000 x7 : ffff8000eec1c000 x6 : 0000000000000001
[    4.524427] x5 : ffff8000eec1c000 x4 : fffffc00000f8af4 x3 : 0000000000000000
[    4.525066] x2 : 0000000000000002 x1 : 0000000000000012 x0 : ffff00000282a4c0
[    4.525707] Call trace:
[    4.525929]  handle_mm_fault+0x218/0x22c
[    4.526286]  do_page_fault+0x1c0/0x3f4
[    4.526631]  do_translation_fault+0x94/0xa0
[    4.527011]  do_mem_abort+0x44/0xb0
[    4.527331]  el1_abort+0x40/0x6c
[    4.527630]  el1h_64_sync_handler+0xa4/0xd0
[    4.528010]  el1h_64_sync+0x78/0x7c
[    4.528327]  __arch_clear_user+0xc/0x70
[    4.528676]  load_elf_binary+0xc44/0x1990
[    4.529042]  bprm_execve+0x1f8/0x4fc
[    4.529369]  do_execveat_common+0x180/0x214
[    4.529747]  __arm64_sys_execve+0x38/0x50
[    4.530112]  invoke_syscall.constprop.0+0x50/0xf0
[    4.530536]  do_el0_svc+0x50/0x140
[    4.530845]  el0_svc+0x18/0x54
[    4.531126]  el0t_64_sync_handler+0xe8/0x114
[    4.531514]  el0t_64_sync+0x184/0x188
[    4.531845] ---[ end trace fd83206a0df46d39 ]---
```
And then, this kind of messages are printed repeatedly, each time it is different.
```
[   72.912045] rcu: INFO: rcu_preempt self-detected stall on CPU
[   72.912584] rcu:     1-....: (6000 ticks this GP) idle=f07/1/0x4000000000000002 softirq=523/527 fqs=2958
[   72.913410]  (t=6001 jiffies g=-315 q=425)
[   72.913777] Task dump for CPU 1:
[   72.914065] task:dnsmasq         state:R  running task     stack:    0 pid: 1341 ppid:     1 flags:0x00000002
[   72.914944] Call trace:
[   72.915162]  dump_backtrace+0x0/0x17c
[   72.915500]  show_stack+0x18/0x40
[   72.915799]  sched_show_task+0x148/0x174
[   72.916152]  dump_cpu_task+0x44/0x58
[   72.916477]  rcu_dump_cpu_stacks+0xe4/0x128
[   72.916849]  rcu_sched_clock_irq+0x98c/0xc00
[   72.917232]  update_process_times+0x9c/0xf0
[   72.917606]  tick_sched_timer+0x58/0xd0
[   72.917952]  __hrtimer_run_queues+0x138/0x1e0
[   72.918340]  hrtimer_interrupt+0xe8/0x244
[   72.918698]  arch_timer_handler_phys+0x34/0x44
[   72.919098]  handle_percpu_devid_irq+0x84/0x130
[   72.919501]  handle_domain_irq+0x60/0x90
[   72.919853]  gic_handle_irq+0x54/0x130
[   72.920192]  call_on_irq_stack+0x20/0x54
[   72.920543]  do_interrupt_handler+0x54/0x60
[   72.920916]  el1_interrupt+0x30/0x80
[   72.921240]  el1h_64_irq_handler+0x18/0x24
[   72.921607]  el1h_64_irq+0x78/0x7c
[   72.921912]  queued_spin_lock_slowpath+0x58/0x2e0
[   72.922334]  __d_lookup+0x78/0x170
[   72.922643]  d_lookup+0x44/0x70
[   72.922928]  path_openat+0x2f8/0xe50
[   72.923247]  do_filp_open+0x70/0x100
[   72.923567]  do_sys_openat2+0x98/0x150
[   72.923906]  __arm64_sys_openat+0x5c/0x80
[   72.924266]  invoke_syscall.constprop.0+0x50/0xf0
[   72.924686]  do_el0_svc+0x50/0x140
[   72.924992]  el0_svc+0x18/0x54
[   72.925269]  el0t_64_sync_handler+0xe8/0x114
[   72.925652]  el0t_64_sync+0x184/0x188
```

msdos03 committed:
rockchip: add FriendlyARM NanoPC T4 support

Hardware
--------
RockChip RK3399 ARM64 (6 cores)
4GB LPDDR3 RAM
1x 1000 Base-T
1 GPIO LED (status)
HDMI 2.0
3.5mm TRRS AV jack
Micro-SD slot
16GB eMMC
1x USB 3.0 Port
2x USB 2.0 Port
1x USB Type-C Port
1x M.2 PCI-E Port
AP6356S (BCM4356) SDIO WiFi & Bluetooth adapter
--------
Note: AP6356S is not supported yet due to the lack of firmware and NVRAM

Signed-off-by: Lu jicong <jiconglu58@gmail.com>

1715173329 commented:
Cc @chunkeey @ynezz

msdos03 commented:
@1715173329 After some more tests, It seems that not gpio-buttons-hotplug causes these errors, I think I'd better do some more work and close this PR
